python/lsst/analysis/tools/actions/plot/histPlot.py

```diff
@@ -216,56 +270,118 @@ def _assignColors(self):
             ],
         )
         if self.cmap in custom_cmaps.keys():
-            all_cols = custom_cmaps[self.cmap]
+            all_colors = custom_cmaps[self.cmap]
         else:
             try:
-                all_cols = getattr(plt.cm, self.cmap).copy().colors
+                all_colors = getattr(plt.cm, self.cmap).copy().colors
             except AttributeError:
                 raise ValueError(f"Unrecognized color map: {self.cmap}")
 
         counter = 0
-        cols = defaultdict(list)
+        colors = defaultdict(list)
         for panel in self.panels:
             for hist in self.panels[panel].hists:
-                cols[panel].append(all_cols[counter % len(all_cols)])
+                colors[panel].append(all_colors[counter % len(all_colors)])
                 counter += 1
-        return cols
+        return colors
 
-    def _makePanel(self, data, panel, ax, col, **kwargs):
+    def _makePanel(self, data, panel, ax, colors, label_font_size=9, legend_font_size=7, ncols=1, **kwargs):
         """Plot a single panel containing histograms."""
-        panel_range = self._getPanelRange(data, panel)
         nums, meds, mads = [], [], []
+        for i, hist in enumerate(self.panels[panel].hists):
+            hist_data = data[hist][np.isfinite(data[hist])]
+            num, med, mad = self._calcStats(hist_data)
+            nums.append(num)
+            meds.append(med)
+            mads.append(mad)
+        if self.panels[panel].doPercentileRange:
+            panel_range = self._getPercentilePanelRange(data, panel)
+        else:
+            # Set the panel range to be extend 3.5 times the maximum sigmaMad
+            # for the datasets in the panel to the left[right] from the
+            # minimum[maximum] median value of all datasets in the panel.
+            maxMad = np.nanmax(mads)
+            maxMed = np.nanmax(meds)
+            minMed = np.nanmin(meds)
+            panel_range = [minMed - 3.5 * maxMad, maxMed + 3.5 * maxMad]
+            if panel_range[1] - panel_range[0] == 0:
+                log.info(
+                    "NOTE: panel_range for {} based on med/sigMad was 0.  Computing using "
+                    "percentile range instead.".format(panel)
+                )
+                panel_range = self._getPercentilePanelRange(data, panel)
```

Comment:
A few comments here:

1) I think much of this `panel_range` logic needs to be moved back into the method previously called `getPanelRange` (renamed here to `getPercentilePanelRange`). Having all the get-my-plotting-range logic in one method makes more sense to my eye. 

2) We probably also want to add an absolute range user input option, in addition to percentile scaling and sigma-mad scaling. This would help us reproduce plots with a fixed range on the x-axis, to aid in cross-comparisons. To achieve this, I'd recommend replacing `doPercentileRange` with a string field named something like `rangeType`. The default for this should be `percent`, with `lower = 0` and `upper = 100` (i.e., plotting everything, which probably makes the most sense as a generic default). This then allows us to think about:

3) I think the sigma-mad range also needs to be configurable. To achieve this here, I would recommend changing `pLower` and `pUpper` to more generic labels, such as `lower` and `upper`. Thus, if the user sets `rangeType` to `sigma`, then `lower` would be the lower sigma range (and similarly for `upper`).